src/app/dashboard/components/dashboard.component.ts

```diff
@@ -29,7 +30,11 @@ export class DashboardComponent implements OnInit, OnDestroy {
 
   handleStatusClick(agreement: Agreement, type: String) {
     this.dashboardService.setSelectedAgreement(agreement);
-    this.router.navigate(['/agreement-detail'], { queryParams: { type }});
+    this.router.navigate(['/agreement-detail'], { queryParams: { type } });
+  }
+
+  formatItemDetails(agreement: Agreement) {
+    return `${agreement.quantity}${agreement.unitOfIssue ? ', ' + agreement.unitOfIssue : ''}${agreement.details ? ', ' + agreement.details : ''}`
```

Comment:
code: how often is this called? If this is bound directly to the view, it could be on a lot fo different dom events (can't recall with ones happen on every mouse move, vs just entry/exit. Anyway, if performance is a concern, consider changing the view change detection strategy to onpush or possible caching this string and just rebuild on data change